zerver/lib/cache.py

```diff
@@ -346,6 +346,12 @@ def delete_user_profile_caches(user_profiles):
 
     cache_delete_many(keys)
 
+def delete_display_recipient_cache(user_profile):
+    # type: (UserProfile) -> None
+    subscriptions = user_profile.subscription_set.select_related()
```

Comment:
Can't this be just `Subscription.objects.filter(user_profile=user_profile).values_list('recipient_id', flat=True)`?  I think that'd be a more efficient query, since we just need to fetch the IDs from the server.